src/devices/uFire_ISE/UFireIse.cs

```diff
@@ -0,0 +1,388 @@
+
+using System;
+using System.Device;
+using System.Device.I2c;
+using UnitsNet;
+
+namespace Iot.Device.UFire
+{
+    /// <summary>
+    /// μFire ISE (Ion Specific Electrode) Probe Interface controller
+    /// </summary>
+    public class UFireIse : IDisposable
+    {
+        private const int ISE_TEMP_MEASURE_TIME = 750;
+        private const int ISE_MV_MEASURE_TIME = 250;
+        private const int ISE_COMMUNICATION_DELAY = 10;
+
+        private const bool ISE_DUALPOINT_CONFIG_BIT = false;
+        private const int ISE_TEMP_COMPENSATION_CONFIG_BIT = 1;
+
+        private I2cDevice _device;
+
+        private float _mV = 0;
+        private Temperature _temp = new Temperature();
+
+        private bool _temperatureCompensation = false;
+
+        /// <summary>
+        /// Use temperature compensation
+        /// </summary>
+        public bool TemperatureCompensation
+        {
+            get
+            {
+                return _temperatureCompensation;
+            }
+
+            set
+            {
+                _temperatureCompensation = value;
+
+                byte retval = Read_byte((byte)Register.ISE_CONFIG_REGISTER);
+
+                retval = (byte)Bit_set((int)retval, ISE_TEMP_COMPENSATION_CONFIG_BIT, _temperatureCompensation);
+
+                Write_byte((byte)Register.ISE_CONFIG_REGISTER, retval);
+            }
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="UFireIse"/> class.
+        /// </summary>
+        /// <param name="i2cDevice">The I2C device to be used</param>
+        public UFireIse(I2cDevice i2cDevice)
+        {
+            _device = i2cDevice;
+        }
+
+        /// <summary>
+        /// Measure mV
+        /// </summary>
+        /// <returns>mV read from ISE Probe Interface</returns>
+        public ElectricPotential MeasuremV()
+        {
+            Send_command((byte)Command.ISE_MEASURE_MV);
+
+            DelayHelper.DelayMilliseconds(ISE_MV_MEASURE_TIME, allowThreadYield: true);
+
+            _mV = Read_register((byte)Register.ISE_MV_REGISTER);
+            _mV = Convert.ToSingle(Math.Round(_mV, 2));
+
+            return new ElectricPotential(_mV, UnitsNet.Units.ElectricPotentialUnit.Millivolt);
+        }
+
+        /// <summary>
+        /// Measure temperature
+        /// </summary>
+        /// <returns>Temperature. Temperature range: -2 to 35 C. Temperature Precision: 0.05 C</returns>
+        public Temperature MeasureTemp()
+        {
+            Send_command((byte)Command.ISE_MEASURE_TEMP);
+
+            DelayHelper.DelayMilliseconds(ISE_TEMP_MEASURE_TIME, allowThreadYield: true);
+            _temp = new Temperature(Read_register((byte)Register.ISE_TEMP_REGISTER), UnitsNet.Units.TemperatureUnit.DegreeCelsius);
+
+            return _temp;
+        }
+
+        /// <summary>
+        /// Set temperature
+        /// </summary>
+        /// <param name="temp">Temperature</param>
+        public void SetTemp(Temperature temp)
```

Comment:
If this is to compensate the temperature or to correct the temperature, then consider using a function with a more explicit name: SetTemperatureCompensation